test/module/aa-module.specs.ts

```diff
@@ -0,0 +1,1057 @@
+            ),
+          ]),
+        },
+        walletOwner,
+        entryPoint
+      );
+      console.log(userOpAA2);
+      await entryPoint.handleOps(
+        [userOpAA2],
+        await offchainSigner.getAddress(),
+        {
+          gasLimit: 10000000,
+        }
+      );
+    });
+
+    it("send transaction from enabled social recovery module using EntryPoint (and Paymaster)", async function () {
+      // Now here also the wallet with owner and index 2 should have been deployed
+
+      const expectedSmartAccountAddress =
+        await walletFactory.getAddressForCounterfactualAccount(owner, 2);
+
+      userSCW = await ethers.getContractAt(
+        "contracts/smart-contract-wallet/SmartAccount.sol:SmartAccount",
+        expectedSmartAccountAddress
+      );
+
+      const prevNonce = await userSCW.nonce();
+      console.log("previous nonce is: ", prevNonce.toNumber());
+
+      await token
+        .connect(accounts[0])
+        .transfer(expectedSmartAccountAddress, ethers.utils.parseEther("100"));
+
+      // module is already deployed, enabled and destination whitelisted
+
+      const isEnabled = await userSCW.isModuleEnabled(
+        socialRecoveryModule.address
+      );
+      expect(isEnabled).to.be.equal(true);
+
+      /* expect(
+        await userSCW.getModulesPaginated(AddressOne, 10)
+      ).to.be.deep.equal([[socialRecoveryModule.address], AddressOne]); */
+
+      // invoking transaction via enabled module
+
+      const SmartAccount = await ethers.getContractFactory("SmartAccount");
+      const SocialRecoveryModule = await ethers.getContractFactory(
+        "SocialRecoveryModule"
+      );
+
+      const newOwner = accounts[5];
+
+      // bob confirms transaction for setOwner()
+      tx = await socialRecoveryModule
+        .connect(accounts[1])
+        .confirmTransaction(userSCW.address, newOwner.address);
+      console.log(await userSCW.owner());
+
+      // charlie confirms transaction for setOwner()
+      tx = await socialRecoveryModule
+        .connect(accounts[2])
+        .confirmTransaction(userSCW.address, newOwner.address);
+
+      // have to make entrypoint do confirmTransaction somehow
+
+      const txnDataModule = SocialRecoveryModule.interface.encodeFunctionData(
+        "recoverAccess",
+        [userSCW.address, newOwner.address]
+      );
+
+      const txnDataAA2 = SmartAccount.interface.encodeFunctionData(
+        "executeCall",
+        [
+          socialRecoveryModule.address,
+          ethers.utils.parseEther("0"),
+          txnDataModule,
+        ]
+      );
+
+      console.log("data for executeCall");
+
+      // const smartAccountCallData = "0x";
+      const userOp1 = await fillAndSign(
+        {
+          sender: expectedSmartAccountAddress,
+          callData: txnDataAA2,
+          // verificationGasLimit: 5000000,
+          // no callGasLImit override as wallet is deployed
+        },
+        walletOwner,
```

Comment:
it should work for any signer, right, since it is the enabled module transaction?

Reply:
yep true, i was thinking to replace the sig with mock sig in final userop, this works too. will replace signer